transmission/transmission_test.go

```diff
@@ -820,6 +823,100 @@ func TestFireBatchWithBrokenFirstEvent(t *testing.T) {
 	})
 }
 
+type fakeBatch struct {
+	block <-chan struct{}
+	send  chan<- []interface{}
+	items []interface{}
+}
+
+func (fb *fakeBatch) Add(item interface{}) {
+	fb.items = append(fb.items, item)
+}
+
+func (fb *fakeBatch) Fire(notifier muster.Notifier) {
+	defer notifier.Done()
+	<-fb.block
+	fb.send <- fb.items
+}
+
+func TestHoneycombTransmissionFlush(t *testing.T) {
+	ev := &Event{
+		Metadata: "adder",
+	}
+
+	t.Run("Flush should send all data that has been Added", func(t *testing.T) {
+		w := new(Honeycomb)
+		w.MaxBatchSize = 1000
+		w.PendingWorkCapacity = 1
+		block := make(chan struct{})
+		close(block) // don't block
```

Comment:
could you add more context here or above in the `fakeBatch` struct definition? creating a channel and immediately closing it seems weird, and I'm guessing it has something to do with how you want the flow of this test to go but I don't quite get it yet.